packages/dashboard-frontend/src/store/Workspaces/index.ts

```diff
@@ -176,10 +183,38 @@ export const actionCreators: ActionCreators = {
   startWorkspace:
     (workspace: Workspace, params?: ResourceQueryParams): AppThunk<KnownAction, Promise<void>> =>
     async (dispatch, getState): Promise<void> => {
+      const openOauthPage = (oauthUrl: string, namespace: string, workspaceName: string) => {
```

Comment:
Don't you mind extracting this function and the similar one from the factory loader ([openAuthPage](https://github.com/eclipse-che/che-dashboard/pull/640/files#diff-497e6c2dff7f716c2f404dcddd0c83667f010ded8ac6eb4474d445dc45eab5dfR207-R225)) into a separate module (service)?

Reply:
done